### DISCUSSION TITLE
Build failure on openSUSE/Tumbleweed

### DISCUSSION THREAD

**Discussion by mcepl** [espanso/espanso#1550] [Q&A]
When building an openSUSE package, compilation of C sources fails, because script is not able to find that `xkbcommon` include files are located in a subdirectory and so C compiler requires `-I/usr/include/libxkbcommon`. I have even added

```spec
export CFLAGS="%{optflags} $(pkgconf --cflags xkbcommon)"
```

to make it obvious for gcc, but it doesn’t help and the build fails with:

```
[  208s]   running: "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "src/evdev" "-Wall" "-Wextra" "-o" "/home/abuild/rpmbuild/BUILD/espanso-2.1.8/target/release/build/espanso-detect-3b5a267d5767cec8/out/src/evdev/native.o" "-c" "src/evdev/native.cpp"
[  208s]   cargo:warning=src/evdev/native.cpp:49:10: fatal error: xkbcommon/xkbcommon.h: No such file or directory
[  208s]   cargo:warning=   49 | #include "xkbcommon/xkbcommon.h"
[  208s]   cargo:warning=      |          ^~~~~~~~~~~~~~~~~~~~~~~
[  208s]   cargo:warning=compilation terminated.
[  208s]   exit status: 1
[  208s] 
[  208s]   --- stderr
[  208s] 
[  208s] 
[  208s]   error occurred: Command "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "src/evdev" "-Wall" "-Wextra" "-o" "/home/abuild/rpmbuild/BUILD/espanso-2.1.8/target/release/build/espanso-detect-3b5a267d5767cec8/out/src/evdev/native.o" "-c" "src/evdev/native.cpp" with args "c++" did not execute successfully (status code exit status: 1).
```

How to persuade c++ that headers are where they are?

[Complete build log](https://github.com/espanso/espanso/files/10999108/_log.txt) with all details.

**Comment by mcepl** [comment]
So, I was on the right path, I only had to add the same line for `CXXFLAGS` to make it to build. Now however I went to another dead end … `espanso` doesn’t seem to be able to find its own code when linking!

See [_log.txt](https://github.com/espanso/espanso/files/11008083/_log.txt) for more.

**Comment by martisj** [comment]
I am seeing the same build error :(
